packages/file-search/src/browser/quick-file-open.ts

```diff
@@ -27,24 +27,26 @@ import * as fuzzy from '@theia/core/shared/fuzzy';
 import { MessageService } from '@theia/core/lib/common/message-service';
 import { FileSystemPreferences } from '@theia/filesystem/lib/browser';
 import { EditorOpenerOptions, EditorWidget, Position, Range } from '@theia/editor/lib/browser';
-import { findMatches, QuickInputService } from '@theia/core/lib/browser/quick-input/quick-input-service';
+import { findMatches, QuickInputService, QuickPickItem, QuickPicks } from '@theia/core/lib/browser/quick-input/quick-input-service';
 
 export const quickFileOpen: Command = {
     id: 'file-search.openFile',
     category: 'File',
     label: 'Open File...'
 };
-
 export interface FilterAndRange {
     filter: string;
     range?: Range;
 }
 
 // Supports patterns of <path><#|:><line><#|:|,><col?>
 const LINE_COLON_PATTERN = /\s?[#:\(](?:line )?(\d*)(?:[#:,](\d*))?\)?\s*$/;
+type FileQuickPickItem = QuickPickItem & { uri: URI };
 
 @injectable()
-export class QuickFileOpenService implements monaco.quickInput.IQuickAccessDataService {
+export class QuickFileOpenService implements QuickAccessProvider {
```

Comment:
I am contemplating the possibility to support file search by text and symbol e.g. 'filex@functiony'
and similarly to `vscode` select a file via the drop down and follow the selection with the `@` symbol.
 
In order to achieve that we would need to use a `provider` interface like `IQuickAccessProvider` or extend `PickerQuickAccessProvider` so we can get a handle to the picker instance `IQuickPickItem` e.g. from the method `provide`. 
Having access to the picker can then make it possible to keep track of the `active` items i.e. the selected item entries in the drop down and therefore be the foundation to navigate to e.g. `filex@symboly`.

What do you think about replacing this interface for one of the options mentioned above?

Reply:
The point of this PR is to hide the implementation details (aka everything imported from VS Code), so no, exposing IQuickAccessProvider or other VS Code API is a no-no, IMO. But nothing prevents us from exposing more functionality in the QuickInput/QuickAccess interfaces. Why don't you open an issue describing the feature?

Reply:
It's fine if we handle the whole feature in a separate PR.
The issue describing the feature already exists, see:
https://github.com/eclipse-theia/theia/issues/9477

Reply:
I believe what you want is already possible: see how the quick pick is manipulated in plugin-ext-deploy-command.ts

Reply:
The reference you provided is nice, although it does not apply to `quick-file-open` where we handle PREFIX='',
for this case and looking at `vscode` as a reference, the implementation relies on registering a provider instance of `IQuickAccessProvider` via `AnythingQuickAccessProvider` so the `provide` method may nicely override the one in the parent `PickerQuickAccessProvider` and therefore gain access to the picker, and when calling  `super.provide` which reaches `PickerQuickAccessProvider` it will then call `getPicks` which can be overridden as well to provide the customized behavior..  

We can leave this refactoring for later although it would be great if we can get your expertise in the area to prepare the foundation for it..